src/Microsoft.ML.KMeansClustering/KMeansPlusPlusTrainer.cs

```diff
@@ -44,13 +45,20 @@ namespace Microsoft.ML.Trainers
     /// | Required NuGet in addition to Microsoft.ML | None |
     ///
     /// ### Training Algorithm Details
-    /// K-means is a popular clustering algorithm. With K-means, the data is clustered into a specified
-    /// number of clusters in order to minimize the within-cluster sum of squares.
+    /// [K-means](https://en.wikipedia.org/wiki/K-means_clustering) is a popular clustering algorithm.
+    /// With K-means, the data is clustered into a specified number of clusters in order to minimize the within-cluster sum of squared distances.
     /// K-means++ improves upon K-means by using the [Yinyang K-Means](https://research.microsoft.com/apps/pubs/default.aspx?id=252149)
     /// method for choosing the initial cluster centers.
-    /// K-Means++ accelerates K-Means up to an order of magnitude while producing exactly the same clustering results(modulo floating point precision issues).
+    /// K-Means++ accelerates K-Means up to an order of magnitude while producing exactly the same clustering results (modulo floating point precision issues).
     /// K-Means++ observes that there is a lot of redundancy across iterations in the KMeans algorithms and most points do not change their clusters during an iteration.
     /// It uses various bounding techniques to identify this redundancy and eliminate many distance computations and optimize centroid computations.
+    ///
+    /// ### Scorning Function
+    /// The output score column contains the L2-norm distance of the given input vector $\textbf{x}\in \mathbb{R}^n$ to each cluster's centroid.
+    /// Assume that the centriod of the $c$-th cluster is $\textbf{m}_c \mathbb{R}^n$.
+    /// The $c$-th value at the Score column would be $d_c = || \textbf{x} - \textbf{m}_c ||_2^2$.
+    /// The predicted label is the index with the smallest value in a $K$ dimension vector $[d_1, \dots, d_K]$, where $K$ is the number of clusters.
```

Comment:
>[d_1, \dots, d_K] [](start = 92, length = 17)

I would write the indices from 0 to K-1.

Reply:
Ok.

---
In reply to: [277753639](https://github.com/dotnet/machinelearning/pull/3520#discussion_r277753639) [](ancestors = 277753639)